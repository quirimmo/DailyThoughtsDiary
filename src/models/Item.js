class Item {
	constructor(date, symptoms, location, thoughts) {
		this.key = undefined;
		this.date = date;
		this.symptoms = symptoms;
		this.location = location;
		this.thoughts = thoughts;
	}
}

export default Item;
