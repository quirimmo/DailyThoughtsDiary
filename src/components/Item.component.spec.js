import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';
import Item from './Item.component';

const item = {
	location: 'location',
	symptoms: 'symptoms',
	thoughts: 'thoughts',
	date: '2018-12-25T14:00:46.723Z'
};
const mockOnDeleteItem = jest.fn();
const mockOnUpdateItem = jest.fn();
let component;
let instance;

describe('Item', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		NavigationTestUtils.resetInternalState();
		component = renderer.create(
			<Item
				item={item}
				onDeleteItem={mockOnDeleteItem}
				onUpdateItem={mockOnUpdateItem}
			/>
		);
		instance = component.getInstance();
	});
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('should be defined', () => {
		expect(Item).toBeDefined();
	});

	it('should render the component', async () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	describe('onDeleteItem', () => {
		it('should call the onDeleteItem prop', () => {
			instance.onDeleteItem();
			expect(mockOnDeleteItem).toHaveBeenCalledWith(item);
		});
	});

	describe('onUpdateItem', () => {
		it('should call the onUpdateItem prop', () => {
			instance.onUpdateItem();
			expect(mockOnUpdateItem).toHaveBeenCalledWith(item);
		});
	});
});
