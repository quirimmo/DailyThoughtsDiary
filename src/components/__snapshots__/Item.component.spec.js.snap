// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render the component 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    symptoms
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    location
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    thoughts
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Tuesday 25-12-2018 15:00
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 20,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Delete Item
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 20,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Update Item
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
